asm_utils/src/reachability.rs

```diff
@@ -168,19 +168,21 @@ pub fn extract_label_offsets<R: Register, F: FunctionOpKind>(
         })
 }
 
-pub fn references_in_statement<R: Register, F: FunctionOpKind>(
+pub fn references_in_statement<R: Register, F: FunctionOpKind, A: Architecture>(
     statement: &Statement<R, F>,
 ) -> BTreeSet<&str> {
     let mut ret = BTreeSet::new();
     match statement {
         Statement::Label(_) | Statement::Directive(_, _) => (),
-        Statement::Instruction(_, args) => {
-            for arg in args {
-                arg.post_visit_expressions(&mut |expr| {
-                    if let Expression::Symbol(sym) = expr {
-                        ret.insert(sym.as_str());
-                    }
-                });
+        Statement::Instruction(name, args) => {
+            if A::instruction_can_have_references(name) {
```

Comment:
I think it would be clearer to return a vector of references. Does it work to have a default implementation in the trait that has more or less this code here, and then in the architecture we call the default implementation only if it is not fence.